framework/src/main/java/org/checkerframework/framework/util/typeinference8/constraint/ConstraintSet.java

```diff
@@ -190,12 +224,17 @@ public ConstraintSet getClosedSubset(Dependencies dependencies) {
       return subset;
     }
 
-    outDependencies.clear();
-    inputDependencies.clear();
+    // TODO: double check that this code is correct.
+    // checker/tests/all-systems/java8inference/MapEntryGetFails.java is a test that uses this code.
+    Set<Variable> inputDependencies = new LinkedHashSet<>();
+    Set<Variable> outDependencies = new LinkedHashSet<>();
     // If this subset is empty, then there is a cycle (or cycles) in the graph of dependencies
```

Comment:
It's true that some cycle exists somewhere.  Is it possible that a closed subset already exists at this moment?  Does it matter?